src/averageAges.js

```diff
@@ -55,7 +74,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child => people.some(mom => child.mother === mom.name
+      && child.sex === 'm'))
+    : people.filter(child => people.some(mom => child.mother === mom.name));
```

Comment:
Try not to duplicate yourself. In both cases, you have almost the same logic. 
Think about using the ternary operator inside the filter callback.
It will help you to improve performance and make your code cleaner
You can fix it in all functions

Reply:
+